### ISSUE TITLE
Request: gdb 14.1 integration

### ISSUE THREAD

**Issue by oscarbg** [skeeto/w64devkit#96]

**Comment by skeeto** [comment]
Thanks for the heads up. I didn't know there was a new release this past 
weekend. I have a seemingly-working update (b0ce23d) which required a fix 
to one of my patches and a workaround for a new GDB build bug. I'll need 
to try it out for awhile before putting it into a release, though. GDB is 
probably the least reliable program included in w64devkit, so I need to 
make sure it's in a usable state. If you want to help, build the gdb-14.1 
branch yourself and give GDB a thorough test drive.

The master branch hasn't been touched in nearly three months (46e4afd), 
and I've been dogfooding that commit daily this whole period. That means 
it's in a great state for a release, so I'm going to make a new w64devkit 
release within the next day or so with GDB 13.1. If 14.1 behaves at least 
as well 13.1, it will make it into the release after this.

**Comment by Peter0x44** [comment]
GDB 14.2 has been released now. Maybe this is worth revisiting.

**Comment by skeeto** [comment]
Thanks for pointing this out, @Peter0x44. I made a gdb-14.2 branch with 
the upgrade, so you can try it for yourself. Unfortunately GDB 14.2 still 
busted on i686-w64-mingw32. I didn't report it upstream, so the GDB devs 
are probably unaware that it's been broken for two releases now.

**Comment by Peter0x44** [comment]
@skeeto have you reported this upstream yet, or should I do it? It seems like it won't be addressed unless that happens, there is no one regularly testing i686-w64-mingw32.

**Comment by skeeto** [comment]
I have not reported this upstream, in part because I haven't figured out 
how to present it in a useful way. Inability to natively build GDB on the 
target in question makes it a lot more complicated. So if you'd like to 
tackle this report, go for it!

**Comment by Peter0x44** [comment]
@skeeto if it helps to investigate any further, I had some upstream discussions:
```
i found the reason, gnulib/config.h does this:
#define __MINGW_USE_VC2005_COMPAT 1
which in turn enables this in _mingw.h:
#define _USE_32BIT_TIME_T
...
so bfd and gdb have different opinions what time_t really is
```

**Comment by Peter0x44** [comment]
bug report: https://sourceware.org/bugzilla/show_bug.cgi?id=31636

**Comment by oscarbg** [comment]
Nice find.. seems either gdb or mingw have to agree.. in the meanwhile possibly a custom gdb patch (to conform to mingw time struct) would be fastest option to fix?

**Comment by Peter0x44** [comment]
@oscarbg you're free to try! I think commenting out the mentioned define in gnulib may work, would be good to get confirmation.

**Comment by Peter0x44** [comment]
Or alternatively, adding the define into CFLAGS manually. That sounds easier than the patch.

**Comment by skeeto** [comment]
Thanks, @Peter0x44! I've updated the gdb-14.2 branch with the CFLAGS and 
CXXFLAGS defines, and it seems to do the trick. It's still unclear to me 
why this suddenly became a problem in 14: gnulib is unchanged between 13 
and 14, and 13 was already using stat.

I'll test drive GDB 14.2 for awhile, especially 32-bit, and merge if it 
works at least as well as 13.1.

**Comment by Peter0x44** [comment]
> it's still unclear to me why this suddenly became a problem in 14
Further details are in the bug report.

**Comment by skeeto** [comment]
Thanks again, @Peter0x44. Amazing how effective it is simply to reach out 
and ask someone more familiar with the problem! (I need to get better at 
that.)

With this fix, GDB 14.2 has been working well for me today. No more 
hiccups. So I'll likely merge that branch soon.
